oci/spec_opts_unix.go

```diff
@@ -478,25 +491,45 @@ func getAllCapabilities() []string {
 
 var errNoUsersFound = errors.New("no users found")
 
-func getUIDGIDFromPath(root string, filter func(user.User) bool) (uid, gid uint32, err error) {
+func getUIDGIDFromPath(root string, filter func(user.User) bool) (uint32, uint32, []uint32, error) {
 	ppath, err := fs.RootPath(root, "/etc/passwd")
 	if err != nil {
-		return 0, 0, err
+		return 0, 0, nil, err
 	}
 	users, err := user.ParsePasswdFileFilter(ppath, filter)
 	if err != nil {
-		return 0, 0, err
+		return 0, 0, nil, err
 	}
 	if len(users) == 0 {
-		return 0, 0, errNoUsersFound
+		return 0, 0, nil, errNoUsersFound
 	}
 	u := users[0]
-	return uint32(u.Uid), uint32(u.Gid), nil
+	gpath, err := fs.RootPath(root, "/etc/group")
+	if err != nil {
+		return 0, 0, nil, nil
```

Comment:
shouldln't we return u.Uid,u.Gid instead of 0, 0 in this case?

Reply:
Since the purpose of retrieving the missing sgids in this function is to fix a bug for both `docker build` and buildkit, I thoguht it should become an `AND` relation with getting UID and GID. IMHO, if the retrieval of sgids fails, still just returning UID and GID seems to be contrary to our purpose.

Reply:
Since the purpose of retrieving the missing sgids in this function is to fix a bug for both docker build and buildkit, I thoguht it should become an AND relation with getting UID and GID. IMHO, if the retrieval of sgids fails, still just returning UID and GID seems to be contrary to our purpose.